app/pages/project/vpcs/internet-gateway-edit.tsx

```diff
@@ -57,84 +62,96 @@ export function EditInternetGatewayForm() {
-            <SideModal.Heading title="Internet Gateway" className="mb-2">
-              Internet Gateway IP Pool
-              {gatewayIpPools && gatewayIpPools.length > 1 ? 's' : ''}
-            </SideModal.Heading>
-            <div className="flex flex-col gap-4">
-              {gatewayIpPools ? (
-                gatewayIpPools.map((gatewayPool) => (
-                  <PropertiesTable key={gatewayPool.id}>
-                    <PropertiesTable.Row label="Name">
-                      {gatewayPool.name}
-                    </PropertiesTable.Row>
-                    <PropertiesTable.Row label="Description">
-                      <DescriptionCell text={gatewayPool.description} />
-                    </PropertiesTable.Row>
-                    <PropertiesTable.Row label="IP Pool">
-                      <IpPoolCell ipPoolId={gatewayPool.ipPoolId} />
-                    </PropertiesTable.Row>
-                  </PropertiesTable>
-                ))
-              ) : (
-                <EmptyCell />
-              )}
-            </div>
-          </div>
-        </div>
-      </SideModal.Body>
-      <SideModal.Footer>
-        <Button variant="ghost" onClick={onDismiss}>
-          Close
-        </Button>
-      </SideModal.Footer>
-    </SideModal>
+    <SideModalForm
+      title="Internet Gateway"
+      formType="edit"
+      resourceName="Internet Gateway"
+      onDismiss={onDismiss}
+      subtitle={
+        <ResourceLabel>
+          <Gateway16Icon /> {internetGateway.name}
+        </ResourceLabel>
+      }
+      form={form}
+      // TODO: pass actual error when this form is hooked up
+      submitError={null}
+      loading={false}
+    >
+      <PropertiesTable key={internetGateway.id}>
+        <PropertiesTable.Row label="Name">{internetGateway.name}</PropertiesTable.Row>
+        <PropertiesTable.Row label="Description">
+          <DescriptionCell text={internetGateway.description} />
+        </PropertiesTable.Row>
+      </PropertiesTable>
+      <Message
+        variant="info"
+        className="text-balance"
+        content={
+          <>
+            This is a read-only copy of this internet gateway. Use the CLI to create and
+            update internet gateways. More functionality for internet gateways will be
+            included in future releases of the Oxide console.
+          </>
+        }
+      />
+      <FormDivider />
+      <SideModal.Heading title="Internet Gateway">IP Addresses</SideModal.Heading>
+
+      {gatewayIpAddresses ? (
+        <Table aria-label="Port filters">
+          <Table.Header>
+            <Table.HeadCell>Name</Table.HeadCell>
+            <Table.HeadCell>Address</Table.HeadCell>
+          </Table.Header>
+          <Table.Body>
+            {gatewayIpAddresses.map((gatewayIpAddress) => (
+              <Table.Row key={gatewayIpAddress.id}>
+                <Table.Cell className="w-1/2">
+                  <Truncate text={gatewayIpAddress.name} maxLength={24} />
```

Comment:
Could get rid of this is we let the table be scrollable, but I think we need an action col on the end to stop it looking funky